src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -9527,6 +9529,123 @@ static switch_status_t sofia_process_proxy_refer(switch_core_session_t *session,
 	return SWITCH_STATUS_FALSE;
 }
 
+static switch_bool_t has_valid_image_media_type(const char *r_sdp)
+{
+	switch_bool_t ret = SWITCH_FALSE;
+	sdp_parser_t *parser = NULL;
+	sdp_session_t *sdp;
+
+	if ((parser = sdp_parse(NULL, r_sdp, (int) strlen(r_sdp), 0))) {
+		if ((sdp = sdp_session(parser))) {
+			sdp_media_t *m;
+			for (m = sdp->sdp_media; m; m = m->m_next) {
+				if (m->m_type == sdp_media_image && m->m_port) {
+					sdp_attribute_t *attr;
+					ret = SWITCH_TRUE;
+					for (attr = m->m_attributes; attr; attr = attr->a_next) {
+						if (!strcasecmp(attr->a_name, "recvonly")) {
+							ret = SWITCH_FALSE;
+							break;
+						}
+					}
+                }
+			}
+		}
+	}
+	if (parser) {
+		sdp_parser_free(parser);
+	}
+	return ret;
+}
+
+void sofia_handle_sip_i_update(nua_t *nua, sofia_profile_t *profile, nua_handle_t *nh, switch_core_session_t *session, sip_t const *sip,
+								sofia_dispatch_event_t *de, tagi_t tags[])
+{
+	const char *r_sdp = NULL;
+	const char *session_id_header = NULL;
+	private_object_t *tech_pvt = NULL;
+	switch_channel_t *channel = NULL;
+	switch_bool_t has_valid_sdp = SWITCH_TRUE;
+
+	if (session && sip && sip->sip_payload && sip->sip_payload->pl_data) {
+		tech_pvt = switch_core_session_get_private(session);
+		r_sdp = sip->sip_payload->pl_data;
+		if (!strcmp(tech_pvt->mparams.remote_sdp_str, r_sdp)) {
+			switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_DEBUG, "Duplicate SDP\n%s\n", r_sdp);
+			has_valid_sdp = SWITCH_FALSE;
+			goto respond;
+		}
+		if (switch_stristr("m=image", r_sdp)) {
+			switch_t38_options_t *t38_options;
+			if (!has_valid_image_media_type(r_sdp)) {
+				has_valid_sdp = SWITCH_FALSE;
+				goto respond;
+			}
+			channel = switch_core_session_get_channel(session);
+
+			t38_options = switch_core_media_extract_t38_options(session, r_sdp);
+			session_id_header = sofia_glue_session_id_header(session, tech_pvt->profile);
+
+			if (!t38_options) {
+				nua_respond(nh, SIP_488_NOT_ACCEPTABLE, TAG_IF(!zstr(session_id_header), SIPTAG_HEADER_STR(session_id_header)), NUTAG_WITH_THIS_MSG(de->data->e_msg), TAG_END());
+				return;
+			}
+
+			switch_core_media_start_udptl(session, t38_options);
+
+			switch_core_media_set_udptl_image_sdp(session, t38_options, 0);
+		} else {
+			int match;
+			tech_pvt = switch_core_session_get_private(session);
+			channel = switch_core_session_get_channel(session);
+			tech_pvt->mparams.remote_sdp_str = switch_core_session_strdup(session, r_sdp);
```

Comment:
We should only set this if we successfully accept their SDP similar to what we did for SWITCH_R_SDP_VARIABLE

Reply:
correct, i wanted to ask you that